libs/core/runtime_local/src/runtime_local.cpp

```diff
@@ -1387,12 +1387,14 @@ namespace hpx {
         threads::thread_init_data data(util::bind(&runtime::run_helper, this,
                                            func, std::ref(result_), true),
             "run_helper", threads::thread_priority::normal,
-            threads::thread_schedule_hint(0), threads::thread_stacksize::large);
+            // prevent main from being executed directly
+            threads::thread_schedule_hint(
+                threads::thread_schedule_hint_mode::thread, 0, false),
+            threads::thread_stacksize::large);
```

Comment:
It's good to be explicit here, so I like that. However, is there any situation where this could be executed directly? This is just creating an HPX thread that will never be associated with a future, right?

Reply:
I made this explicit as things were failing when running on the shared_priority_scheduler otherwise.

Reply:
That sounds worrying. I hope that was before you made the shared priority scheduler not support direct execution? But even then, if this made things hang the hint itself must've affected something that it shouldn't have affected, since the direct execution should only come in effect with futures, right?

Reply:
I'll revisit this again. This change might not be needed anymore.

Reply:
Looks like this change is not needed anymore (because of: https://github.com/STEllAR-GROUP/hpx/pull/5467/files#diff-31312dbe78048647c045b2467636f2995665d13ba2ec641c0554ca1ede9bddb0R613-R618). I'll revert it.